pipeline-model-definition/src/main/java/org/jenkinsci/plugins/pipeline/modeldefinition/agent/impl/DockerPipeline.java

```diff
@@ -38,12 +38,42 @@
     private String label;
     private String image;
     private String args = "";
+    private String registryUrl;
+    private String registryCredentialsId;
+    private boolean reuseNode;
```

Comment:
We talked this out - I really don't want to default to `true` since that would be a change from existing behavior and could theoretically break existing pipelines.